### PR TITLE
[UMD] Fixed T6832

### PR DESCRIPTION
**Pull request by clayreimann**
https://github.com/babel/babel/pull/3175 fixed up and with tests

### CODE REVIEW

**Comment by clayreimann** [comment]
@hzoo Does this address the outstanding issues?

File: packages/babel-plugin-transform-es2015-modules-umd/src/index.js [hunk 1]
```diff
@@ -39,7 +39,7 @@ export default function ({ types: t }) {
 
     visitor: {
       Program: {
-        exit(path) {
+        exit(path, {opts: {globals: browserGlobals = {}}}) {
```
Comment (hzoo) [inline packages/babel-plugin-transform-es2015-modules-umd/src/index.js]: I think it was more clear without the destructuring?

**Comment by hzoo** [comment]
Thanks for adding the test

**Comment by hzoo** [comment]
Maybe @thejameskyle since you commented on the issue

**Comment by hzoo** [comment]
@clayreimann Can you squash your commits?

File: packages/babel-plugin-transform-es2015-modules-umd/src/index.js [hunk 2]
```diff
@@ -39,7 +39,7 @@ export default function ({ types: t }) {
 
     visitor: {
       Program: {
-        exit(path) {
+        exit(path, plugin) {
```
Comment (thejameskyle) [inline packages/babel-plugin-transform-es2015-modules-umd/src/index.js]: Can you rename `plugin` `state`, just to be consistent with how this is named elsewhere

**Comment by thejameskyle** [comment]
Looks good to me other than that one comment and a squash

**Comment by clayreimann** [comment]
@hzoo @thejameskyle Updated and squashed

**Comment by hzoo** [comment]
Awesome will merge when CI is done

**Pull request merged**